ts/components/Pinpad/index.tsx

```diff
@@ -144,13 +157,56 @@ class Pinpad extends React.PureComponent<Props, State> {
 
   constructor(props: Props) {
     super(props);
-
+    this.placeholderPositions = [...new Array(PIN_LENGTH)];
     this.state = {
       value: "",
-      isDisabled: false
+      isDisabled: false,
+      pinLength: PIN_LENGTH,
+      pinPadValue: new Array("1", "2", "3", "4", "5", "6", "7", "8", "9"),
+      scalableDimension: undefined
     };
+  }
 
-    this.placeholderPositions = [...new Array(PIN_LENGTH)];
+  public componentWillMount() {
+    const { pinPadValue, pinLength } = this.state;
+
+    // tslint:disable-next-line: readonly-array
+    const newPinPadValue = (pinPadValue as string[]).sort(() => {
+      return Math.random() - 0.5;
+    });
+
+    if (
+      (this.props.compareWithCode &&
+        this.props.compareWithCode.length !== PIN_LENGTH) ||
+      this.props.compareWithCode === undefined
+    ) {
+      // tslint:disable-next-line: no-object-mutation
+      this.placeholderPositions = [...new Array(NEW_PIN_LENGTH)];
```

Comment:
You can remove this.
In render method, where it is used, you can replace with this

```typescript
{range(0, this.state.pinLength).map(this.renderPlaceholder)}
```